### PR TITLE
feat: Add agents configuration endpoint with Nexus integration

### PR DESCRIPTION
**Pull request by paulobernardoaf**
- Feature
    - Implement `api/v1/agents` endpoint for processing and uploading skill configurations
    - Add support for streaming responses during agent processing
    - Create utility functions for skill file extraction and processing
    - Integrate with Nexus client for agent upload
    - Add comprehensive logging and error handling
    - Support dynamic skill packaging and entrypoint generation
    - Include environment configuration for Nexus base URL
- Tests
    - Add pytest-asyncio dependency to poetry.lock and pyproject.toml
    - Configure pytest to use asyncio mode automatically
    - Enable async testing capabilities for the project
    - Implement detailed test coverage for agents configuration API
    - Add test cases for successful agent processing, error scenarios, and edge cases
    - Cover streaming response handling, skill extraction, and Nexus integration
    - Test helper functions for skill file processing and content reading
    - Verify error handling, progress tracking, and final success message generation
    - Ensure robust testing of agent configuration workflow

### CODE REVIEW

File: app/api/v1/routers/agents.py [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+                },
+                "success": False,
+                "code": "PROCESSING_ERROR",
+            }
+            yield send_response(error_data, request_id=request_id)
+
+    return StreamingResponse(response_stream(), media_type="application/x-ndjson")
+
+
+# Utility functions below this point
```
Comment (lucaslinhares) [inline app/api/v1/routers/agents.py:166]: unnecessary comment

**Review by lucaslinhares — APPROVED** [review]
